### PR TITLE
support for swift 3

### PR DESCRIPTION
**Pull request by JackieQi**
Tried to convert sample code to swift 3.

### CODE REVIEW

File: OOD-Principles-In-Swift.playground/contents.swift [hunk 1]
```diff
@@ -212,7 +212,7 @@ protocol LandingSiteHaving {
 
 // I can land on LandingSiteHaving objects.
 protocol Landing {
-    func landOn(on: LandingSiteHaving) -> String
+    func landOn(_ on: LandingSiteHaving) -> String
```
Comment (ochococo) [inline OOD-Principles-In-Swift.playground/contents.swift]: `land(on: LandingSiteHaving`

File: OOD-Principles-In-Swift.playground/contents.swift [hunk 2]
```diff
@@ -268,11 +268,11 @@ Example:
 */
 
 protocol TimeTraveling {
-    func travelInTime(time: NSTimeInterval) -> String
+    func travelInTime(_ time: TimeInterval) -> String
```
Comment (ochococo) [inline OOD-Principles-In-Swift.playground/contents.swift]: `travel(time: TimeInterval)` ?

File: README.md [hunk 3]
```diff
@@ -5,15 +5,13 @@ import Foundation
...
-The Principles of OOD in Swift 2.2
+The Principles of OOD in Swift 3.0
 ==================================
 
-A short cheat-sheet with Xcode 7.3 Playground ([OOD-Principles-In-Swift.playground.zip](https://raw.githubusercontent.com/ochococo/OOD-Principles-In-Swift/master/OOD-Principles-In-Swift.playground.zip)).
+A short cheat-sheet with Xcode 8.1 Playground ([OOD-Principles-In-Swift.playground.zip](https://raw.githubusercontent.com/ochococo/OOD-Principles-In-Swift/master/OOD-Principles-In-Swift.playground.zip)).
 
 👷 Project maintained by: [@nsmeme](http://twitter.com/nsmeme) (Oktawian Chojnacki)
 
-⚠️ See my most popular project to date: [Design-Patterns-In-Swift](https://github.com/ochococo/Design-Patterns-In-Swift)
```
Comment (ochococo) [inline README.md:15]: 😋 Okaaaay 😋

File: README.md [hunk 4]
```diff
@@ -177,41 +175,41 @@ Example:
...
 }
 
 // I don't know what RequestError is and will fail and return a NSError.
-func willReturnObjectOrError() -> (object: AnyObject?, error: NSError?) {
-
-    let request = NSURLRequest()
-    let result = fetchData(request)
-
-    return (result.data, result.error)
+func willReturnObjectOrError(from URL: URL) -> (object: Any?, error: NSError?) {
```
Comment (ochococo) [inline README.md]: `(from url: URL)`?

**Review by ochococo — CHANGES_REQUESTED** [review]
Thanks for contributing <3 

Some issues were found.

File: README.md (hunk 3 above)
Comment (JackieQi) [reply README.md:15]: Didn't mean to do that. sorry.

File: README.md (hunk 4 above)
Comment (JackieQi) [reply README.md]: URLRequest doesn't take optional URL. so add let check in new commit

File: OOD-Principles-In-Swift.playground/contents.swift (hunk 2 above)
Comment (JackieQi) [reply OOD-Principles-In-Swift.playground/contents.swift]: I add `in` in parameter which is more consistent with old API i think.

**Comment by naeemshaikh90** [comment]
@ochococo Whats the current status for this? Are you going to merge this PR?

**Comment by ochococo** [comment]
outdated

**Pull request closed without merging**